src/stores/DAOEntityListStore.js

```diff
@@ -0,0 +1,114 @@
+// @flow
+
+import type { DAO, QueryOptions } from 'brewskey.js-api';
+
+import { action, observable } from 'mobx';
+import flattenArray from 'array-flatten';
+import DAOApi, { LoadObject } from 'brewskey.js-api';
+import { createRange } from '../utils';
+
+export type Row<TEntity> = {|
+  key: number,
+  value: LoadObject<TEntity>,
+|};
+
+class DAOEntityListStore<TEntity> {
+  _baseQueryOptions: QueryOptions = {};
+  _dao: DAO<TEntity, *>;
+  _daoSubscriptionID: string;
+  _pageSize: number;
+
+  @observable _queryOptionsList: Array<QueryOptions> = [];
+
+  @observable rows: Array<Row<TEntity>> = [];
+
+  constructor(
+    dao: DAO<TEntity, *>,
+    queryOptions?: QueryOptions = {},
+    pageSize?: number = 20,
+  ) {
+    this._baseQueryOptions = queryOptions;
+    this._dao = dao;
+    this._pageSize = pageSize;
+
+    this._daoSubscriptionID = this._dao.subscribe(this._computeRows);
+    this.reset();
+  }
+
+  dispose = () => {
+    this._dao.unsubscribe(this._daoSubscriptionID);
+  };
+
+  @action
+  fetchNextPage = () => {
+    const currentQuery = this._queryOptionsList[
+      this._queryOptionsList.length - 1
+    ];
+
+    // prevent fetch when we reached end of the list
+    const currentQueryLoadObject = this._dao.fetchMany(currentQuery);
+    if (
+      currentQueryLoadObject.hasValue() &&
+      currentQueryLoadObject.getValueEnforcing().length < this._pageSize
+    ) {
+      return;
+    }
+
+    this._queryOptionsList.push({
+      ...currentQuery,
+      skip: currentQuery.skip + this._pageSize,
+    });
+    this._computeRows();
+  };
+
+  @action
+  reset = () => {
+    // todo when we use it with onRefresh in FlatList
+    // it works normal but doesn't fit standard mobile refresh behavior.
+    // instead of flushing all the cache we need somehow refetch only
+    // the first query and shift all other cache if there are new entitites,
+    // but it seems too complex.
+    this._dao.flushCache();
```

Comment:
If you want this behavior you need to do:
```
const pagesLoadedCount = this._queryOptionsList.length;
this._queryOptionsList = [];
this._queryOptionsList.push({
  ...this._baseQueryOptions,
  skip: 0,
  take: this._pageSize,
});
// There could be a case where an item was deleted between refresh
// so you'll need to do an additional check. If you have the total count
// that will make things easier.
while (this._queryOptionsList.length < pagesLoadedCount) {
  this.fetchNextPage();
}
```

Reply:
but if we don't flush the dao cache here, all that code will just rerender full list with cached values.
If we flush cache, this will be the same as my current solution, but it also refetch all pages that I scrolled before, so idk if this makes sense.

Reply:
Call `flushQueryCache` then. I don't think you need to refetch all the items right?

Also -- reset should only ever be called when the user refreshes the list.  DAO should handle refreshing the caches on mutations.